File: pset6 Python/readability/readability.py
from cs50 import get_string
import string
import re

#Get the text input and set the initial count values
text = get_string("Text: ")
ltrs = 0
wrds = 1
snts = 0


#Count letters
#For characters in text if alphabetic then increment count
for letter in text:
    if letter.isalpha() == True:
        ltrs += 1


#Count words
#Remove all selected punctuation
textFiltered = text + ' '
regex = re.compile("[,\.!?']")
textFiltered = regex.sub('', textFiltered)

#For all characters in text if text is a space increment count
for i in range(len(textFiltered)):
    if (text[i].isspace()):
        wrds += 1



#Count sentences
#For all characters in text if character is end of sentence punctuation then increment count
for letter in text:
    if letter == '.' or letter == '?' or letter == '!':
        snts += 1



#Calculate values of L and S
L = 100 * ltrs/wrds
S = 100 * snts/wrds

#Calcuate the index
index = 0.0588 * L - 0.296 * S - 15.8

#Calculate the grade
grade = round(index)


#Check and output the grade with bounds edited
if grade <= 0:
    print("Before Grade 1\n")
elif grade >= 16:
    print("Grade 16+\n")
elif grade > 0:
    print("Grade", grade,"\n")

File: pset2 Arrays/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int i = 0;
    float ltrs = 0, wrds = 0, snts = 0;



    //Count Letters
    for (i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            ltrs++;

        }
    }
    //printf("Letters: %f\n", ltrs);



    //Count Words
    for (i = 0; text[i] != '\0'; i++)
    {
        if (isspace(text[i]))
        {
            if (!(isspace(text[i+1])))
            {
                wrds++;

            }
        }

    }
    wrds++;
    //printf("Words: %f\n", wrds);



    //Count Sentences
    for (i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            snts++;

        }

    }
    //printf("Sentences: %f\n", snts);



    float L = 100 * ltrs/wrds, S = 100 * snts/wrds;
    //printf("%f\n", L);
    //printf("%f\n", S);

    double index = 0.0588 * L - 0.296 * S - 15.8;

    int grade = round(index);
    //printf("%f\n", index);
    //printf("%d\n", grade);

    if (grade <= 0)
    {
        printf("Before Grade 1\n");

    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");

    }
    else if (grade > 0)
    {
        printf("Grade ""%d\n", grade);

    }


}